src/shared/logs-show.c

```
@@ -333,8 +334,39 @@ static int output_timestamp_monotonic(FILE *f, sd_journal *j, const char *monoto
         if (r < 0)
                 return log_error_errno(r, "Failed to get monotonic timestamp: %m");
 
-        fprintf(f, "[%5"PRI_USEC".%06"PRI_USEC"]", t / USEC_PER_SEC, t % USEC_PER_SEC);
-        return 1 + 5 + 1 + 6 + 1;
+        snprintf(buf, sizeof(buf), "%s", "[");
+        snprintf(buf + strlen(buf), sizeof(buf), "%5"PRI_USEC".%06"PRI_USEC"", t / USEC_PER_SEC, t % USEC_PER_SEC);
+
+        if (mode == OUTPUT_SHORT_DELTA) {
+                sd_id128_t previous_boot_id;
+                uint64_t previous_t, delta;
+
+                r = sd_journal_previous(j);
+                if (r <= 0)
+                        delta = 0;
+                else {
+                        r = sd_journal_get_monotonic_usec(j, &previous_t, &previous_boot_id);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get monotonic timestamp: %m");
+
+                        r = sd_journal_next(j);
+                        if (r <= 0)
+                                /* This shouldn't happen. We just came from this. */
+                                return log_debug_errno(SYNTHETIC_ERRNO(ENODATA), "Whoopsie! We cannot go back to our previously used journal entry.");
```

Please propagate the original error code on `r < 0`, and use synthetic errno on `r == 0`.

---

Done